lxd-agent/devlxd.go

```diff
@@ -51,23 +50,45 @@ type devLxdHandler struct {
 	f func(d *Daemon, w http.ResponseWriter, r *http.Request) *devLxdResponse
 }
 
+func getVSockServer(d *Daemon) (lxd.InstanceServer, error) {
+	// Try connecting to LXD server.
+	client, err := getClient(int(d.serverCID), int(d.serverPort), d.serverCertificate)
+	if err != nil {
+		return nil, err
+	}
+
+	server, err := lxd.ConnectLXDHTTP(nil, client)
+	if err != nil {
+		return nil, err
+	}
+
+	return server, nil
+}
+
 var devlxdConfigGet = devLxdHandler{"/1.0/config", func(d *Daemon, w http.ResponseWriter, r *http.Request) *devLxdResponse {
-	data, err := ioutil.ReadFile("instance-data")
+	server, err := getVSockServer(d)
```

Comment:
getVsockServerClient() would be clearer here, as `server` isn't a vsock server, but a client to the vsock server running in LXD.